### PR TITLE
Create build.yml

### PR DESCRIPTION
**Pull request by portlek**
Part of #81, resolves pr verifying.

### CODE REVIEW

**Comment by MrIvanPlays** [comment]
please make it do so also on master branch commits. thanks

**Comment by portlek** [comment]
@MrIvanPlays i thought if we want to automate the releasing via github actions or codemc things we should separate snapshot workflow from pr verifying.

**Comment by MrIvanPlays** [comment]
> @MrIvanPlays i thought if we want to automate the releasing via github actions or codemc we should separate snapshoting/releasing workflows from pr verifying.

we don't need to. codemc is completely separate from github builds.

**Review by MrIvanPlays — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-java@v1
```
Comment (Jikoo) [inline .github/workflows/build.yml:16]: v2 is current, not v1. Should probably also just use current JDK  (unless we want to set up a matrix for multi-version testing later), it's perfectly capable of targeting older versions.

```yaml
      - name: Set up JDK
        uses: actions/setup-java@v2
        with:
          distribution: 'adopt'
          java-version: '17'
```

**Review by Jikoo — CHANGES_REQUESTED** [review]
Should be using cache action to cache Maven repo, speeds up builds tremendously (and reduces load on others' servers) if the repo is reasonably active.

Sample step:
```yaml
      - name: Cache Maven packages
        uses: actions/cache@v2
        with:
          path: ~/.m2/repository
          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
          restore-keys: |
            ${{ runner.os }}-maven-
```

**Review by Jikoo — APPROVED** [review]
LGTM 👍

**Pull request merged**